### DISCUSSION TITLE
Can't update to 2.17-Beta3 version

### DISCUSSION THREAD

**Discussion by daneel6672** [ThioJoe/YT-Spammer-Purge#947] [Q&A]
Hi,

I'm not familiar with git and so I have to use option 9 to update. I get this 404 error:

```
Getting info about latest updates...
------------------------------------------------------------------------------------------
 A new beta version is available! Visit TJoe.io/latest to see what's new.
   > Current Version: 2.17.0-Beta2
   > Latest Version: 2.17-Beta3
(To stop receiving beta releases, change the 'release_channel' setting in the config file)
------------------------------------------------------------------------------------------

Update Now? (y/n): y

> Downloading version: 2.17-Beta3
Downloading of new version failed!

> Error: GitHub returned a non 200 status code while trying to download newer version.
Status returned: 404
```

How can I update to 2.17-Beta3?

Thanks!

Ramin

**Comment by ThioJoe** [comment]
Hm that's odd, it worked for me. Does it still give the error if you run it now?

Otherwise here is the direct link to the Beta3 download: https://github.com/ThioJoe/YT-Spammer-Purge/releases/download/v2.17.0-Beta3/YTSpammerPurge.2.17.0-Beta3.exe

**Reply by daneel6672** [reply]
Same error - I use Ubuntu so the EXE won't work. What's the URL to get the Beta3 download on linux?

**Reply by ThioJoe** [reply]
Ah there must be an issue with the updater for the python version. You can download the source code here: https://github.com/ThioJoe/YT-Spammer-Purge/archive/refs/tags/v2.17.0-Beta3.zip

Then just extract it and overwrite everything your current directory with the new stuff. It shouldn't overwrite your config files or anything else user made, but maybe make a backup copy of the folder just in case.

**Comment by daneel6672** [comment]
Thanks that worked. Unfortunately the comment from "karen marie emma" here

https://www.youtube.com/watch?v=3-lCBTU7f84&lc=UgxAzimHQLx3imIKNI54AaABAg

still isn't being picked up.

**Reply by ThioJoe** [reply]
Huh odd, you're right

Edit: Ok i figured out the reason, it's because of one of the checks meant to avoid false positives. I'll have to adjust it.

**Comment by ThioJoe** [comment]
This seems to be an issue specifically when running on linux. I have no idea why it causes GitHub to return a 404 though.

However, in the [latest commit](https://github.com/ThioJoe/YT-Spammer-Purge/commit/58cc116f59b3c3a1596e2c0cb75be0a10cab860f) in the beta branch, I've change it so it will retry using an alternative download method.

If you're still having the issue try running it with the latest files from the [beta branch](https://github.com/ThioJoe/YT-Spammer-Purge/tree/2.17-Beta).

If it's still broken, post the screenshot, because the new function will at least show the URL that got sent so I can check that's at least correct.
